Mbed TLS no longer has a development-psa branch

gilles-peskine-arm opened the pull request:
https://os.mbed.com/docs/mbed-os/v6.2/porting/using-psa-enabled-mbed-tls.html points to the `development-psa` branch of Mbed TLS. This was a temporary feature branch. The feature has long been merged into mainstream Mbed TLS. Update instructions accordingly.

This patch should also be made to the documentation for older versions of Mbed OS.

gilles-peskine-arm committed:
PSA is in the main branch of Mbed TLS

The development-psa branch is an old feature branch that has long been
merged into development and master. Refer users to master, which is
the latest release.

Also remove the reference to a submodule. Mbed TLS no longer has any
submodules.

gilles-peskine-arm committed:
Fix formatting glitch

gilles-peskine-arm committed:
When building Mbed TLS, only build the library

Running the tests won't work on a cross build.

danh-arm reviewed (approved):
Looks good. Thanks for doing this.

gilles-peskine-arm committed:
In Mbed TLS, config.pl has been replaced by config.py

config.pl still exists but it only executes config.py. Change the
documentation to use config.py which is more future-proof and avoids
introducing a dependency on Perl.

iriark01 commented:
Thank you!

iriark01 commented:
Published: 5.12 to 6.2
